Use the useLocation hook in Profile2Wrapper

erikphansen opened the pull request:
## Description
So the component can easily see which route we are currently on and it
can update the breadcrumbs accordingly

Using the useLocation hook broke the existing unit tests that used
shallow mounting. I ran into errors using enzyme's full mount as well.
Specifically the useEffect hook used in the Profile2Wrapper's child
ProfileSideNav was causing enzyme.mount() to blow up. But the component
mounts just fine with React Testing Library.

Full error I saw when trying to mount with enzyme:
```
Error: An error was thrown inside one of your components, but React doesn't know what it was. This is likely due to browser flakiness. React does its best to preserve the "Pause on exceptions" behavior of the DevTools, which requires some DEV-mode only tricks. It's possible that these don't work in your browser. Try triggering the error in production mode, or switching to a modern browser. If you suspect that this is actually an issue with React, please file an issue.
```

## Testing done
Local. The final piece of the breadcrumbs work again!

## Screenshots
![ScreenFlow](https://user-images.githubusercontent.com/20728956/84300654-67a1d000-ab07-11ea-8a3e-c0bb12f65fa6.gif)


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

sanlouise reviewed (approved):
Fantastic you got RTL to work! It's great we can now use the `getLocation()` hook instead of passing down the prop 👍

erikphansen committed:
Use the useLocation hook in Profile2Wrapper

So the component can easily see which route we are currently on and it
can update the breadcrumbs accordingly

Using the useLocation hook broke the existing unit tests that used
shallow mounting. I ran into errors using enzyme's full mount as well.
Specifically the useEffect hook used in the Profile2Wrapper's child
ProfileSideNav was causing enzyme.mount() to blow up. But the component
mounts just fine with React Testing Library.

Full error I saw when trying to mount with enzyme:
```
Error: An error was thrown inside one of your components, but React doesn't know what it was. This is likely due to browser flakiness. React does its best to preserve the "Pause on exceptions" behavior of the DevTools, which requires some DEV-mode only tricks. It's possible that these don't work in your browser. Try triggering the error in production mode, or switching to a modern browser. If you suspect that this is actually an issue with React, please file an issue.
```

ncksllvn commented on package.json:
```diff
@@ -121,6 +121,7 @@
     "@department-of-veterans-affairs/generator-vets-website": "^3.3.5",
     "@octokit/rest": "^16.43.1",
     "@sentry/browser": "^5.4.0",
+    "@testing-library/react": "^10.2.1",
```
I just started using testing-library the other day and it's by far my favorite way to write tests now
